core/src/main/java/org/ironjacamar/core/api/bootstrapcontext/CloneableBootstrapContext.java

```diff
@@ -0,0 +1,105 @@
+/*
+ *  IronJacamar, a Java EE Connector Architecture implementation
+ *  Copyright 2016, Red Hat Inc, and individual contributors
+ *  as indicated by the @author tags. See the copyright.txt file in the
+ *  distribution for a full listing of individual contributors.
+ *
+ *  This is free software; you can redistribute it and/or modify it
+ *  under the terms of the Eclipse Public License 1.0 as
+ *  published by the Free Software Foundation.
+ *
+ *  This software is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the Eclipse
+ *  Public License for more details.
+ *
+ *  You should have received a copy of the Eclipse Public License
+ *  along with this software; if not, write to the Free
+ *  Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ *  02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.ironjacamar.core.api.bootstrapcontext;
+
+import org.ironjacamar.core.api.workmanager.WorkManager;
+
+import javax.resource.spi.BootstrapContext;
+import javax.resource.spi.ResourceAdapter;
+import javax.resource.spi.XATerminator;
+import javax.transaction.TransactionSynchronizationRegistry;
+
+/**
+ * The cloneable bootstrap context interface which defines
+ * the contract for all BootstrapContext implementations
+ * @author <a href="mailto:jesper.pedersen@ironjacamar.org">Jesper Pedersen</a>
+ */
+public interface CloneableBootstrapContext extends Cloneable, BootstrapContext
```

Comment:
This should be renamed to BootstrapContext and contain the public API.

E.g. no Cloneable, no set*, ... However, getValidatorFactory() should be added